### PR TITLE
Turret sim

### PR DESCRIPTION
**Pull request by REVSIX**
## Who worked on this PR?
Raunak, Omkaar, Levi, Avyan

## Summary
This PR introduces a full turret simulation layer in MapleSim that allows us to test turret control logic, setpoint tracking, and rotational constraints entirely in simulation.

Before this, turret logic could only be validated on hardware that isn't' there yet. That meant:

- Programming in theory, no test interface for turret and its autoaim capabilities.
- No way to test code logic without robot
- No reproducible edge-case testing

Now:

- Turret logic can be validated off-robot
- Future subsystems (hood, vision, odometry) can integrate into a unified shooting simulation pipeline
- This is the foundation for a fully simulated scoring system.

## Changes Made

- Modified TurretAngleAim.java to work with the simulation abstraction layer.
- Added a TurretIOSim.java file (without physics right now) that relays the turret state in simulation for use in commands.
- Added a ShootAtAngleSim.java file that takes everything from ShooterIOSim away (creating projectiles, which wasn't a physical property of the ShooterIOSim.java) to add to a file that simulates projectiles and the Robot Turret logic and Shooter Logic.
- Modified RobotContainer.java to feed a sim pose and real pose depending on robot state. This will be changed in a future update with vision simulation to use one unified drivetrain variable for updating odometry.
    - Also, initialized turret pose tracking on robot startup rather than through pressing a button, as it was done previously.

## Verification
We ran the following tests:

- Shoot at different points and verify trajectory is going towards the hub.
- Shoot on the move and see if the trajectory keeps up.
- Shoot while rotating to ensure the robot shot trajectory stays locked on the hub.

The results were as follows:

- Shoot at different points ✅ (49/50 shots made while also driving at a slow speed at points appropriate with the fixed hood angle and rpm) : Worked correctly and as intended. Though in the video given below, shots are taken from close to the hub to ensure field state and shot trajectories register towards the simulation game score, a future PR will address shooting from a distance by adjusting RPM and Hood Angle.

- Shoot on the move and see if the trajectory keeps up ❌ : (17/50 shots made while driving at a faster speed at points appropriate with the fixed hood angle and rpm) :  Although it works fine when driving at slow speeds, to work better for a proper shoot on the move we will need to add mathematical functions to address it properly.

- Shoot while rotating to ensure the robot shot trajectory stays locked on the hub. ✅ (50/50 shots made) : Works as intended.

## Verification Videos
[Shoot at different points](https://photos.app.goo.gl/7wZ9TCRgtHk2bsP17)
[Shoot on move](https://photos.app.goo.gl/KLSpNwHnujLYiF2t6)
[Shoot while rotating](https://photos.app.goo.gl/J4ALCZ3hBbB98F4X6)

## Notes for Reviewers
Note: The Robot CAD in the attached videos in incorrectly configured. It rotates around the edge thinking it is the center as a result. For this reason, assume all shots are being shot from the center of the robot. The intake is also inverted as a result. Adjustable hood angle and RPM is also a todo.

### CODE REVIEW

File: src/main/java/frc/robot/commands/ShootAtAngleSim.java [hunk 1]
```diff
@@ -78,9 +83,12 @@ public void shootBalls() {
...
     return new RebuiltFuelOnFly(
         simDrive.getSimulatedDriveTrainPose().getTranslation(),
-        new Translation2d(0, 0), // shooter offet from center
+        new Translation2d(0, 0), // shooter offet from center,
         simDrive.getDriveTrainSimulatedChassisSpeedsFieldRelative(),
-        simDrive.getSimulatedDriveTrainPose().getRotation(),
+        new Rotation2d(
+            simDrive.getSimulatedDriveTrainPose().getRotation().getRadians()
+                + turret.getTurretPositionRadians()
+                + Math.PI), // accounting for drivertrain being flipped?
```
[the remark this answers is not in the document]
Comment (Omkaarcodes) [reply src/main/java/frc/robot/commands/ShootAtAngleSim.java]: +1

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -306,37 +311,27 @@ private void configureButtonBindings() {
...
+                .alongWith(new WaitCommand(1 / 6.0))
+                .repeatedly());
+
+    driverController
+        .y()
+        .whileTrue(
+            Commands.runOnce(() -> simShooter.setLaunchAngle(Units.degreesToRadians(55)))
+                .andThen(() -> simShooter.shootBalls())
+                .alongWith(new WaitCommand(1 / 6.0))
+                .repeatedly());
```
[the remark this answers is not in the document]
Comment (Omkaarcodes) [reply src/main/java/frc/robot/RobotContainer.java]: +1 make sure that the command names are the same for real and sim within the IO structure
